metaflow/client/core.py

```diff
@@ -1525,6 +1541,39 @@ def _log_size(self, stream, meta_dict):
             ds_type, ds_root, stream, attempt, *self.path_components
         )
 
+    def __iter__(self) -> Iterator[DataArtifact]:
```

Comment:
These functions were added to specialize the return type. Exact same functionality (they call their super function)